src/python_testing/TC_DeviceBasicComposition.py

```diff
@@ -0,0 +1,397 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import base64
+import copy
+import json
+import logging
+import pathlib
+import sys
+from dataclasses import dataclass
+from pprint import pprint
+from typing import Any, Callable, Optional
+
+import chip.clusters as Clusters
+import chip.tlv
+from chip import discovery
+from chip.clusters.Attribute import ValueDecodeFailure
+from chip.exceptions import ChipStackError
+from chip.setup_payload import SetupPayload
+from matter_testing_support import AttributePathLocation, MatterBaseTest, async_test_body, default_matter_test_main
+from mobly import asserts
+
+
+def MatterTlvToJson(tlv_data: dict[int, Any]) -> dict[str, Any]:
+    """Given TLV data for a specific cluster instance, convert to the Matter JSON format."""
+
+    matter_json_dict = {}
+
+    key_type_mappings = {
+        chip.tlv.uint: "UINT",
+        int: "INT",
+        bool: "BOOL",
+        list: "ARRAY",
+        dict: "STRUCT",
+        chip.tlv.float32: "FLOAT",
+        float: "DOUBLE",
+        bytes: "BYTES",
+        str: "STRING",
+        ValueDecodeFailure: "ERROR",
+        type(None): "NULL",
+    }
+
+    def ConvertValue(value) -> Any:
+        value_type = type(value)
+
+        if value_type == ValueDecodeFailure:
+            raise ValueError(f"Bad Value: {str(value)}")
+
+        if value_type == bytes:
+            return base64.b64encode(value).decode("UTF-8")
+        elif value_type == list:
+            for idx, item in enumerate(value):
+                value[idx] = ConvertValue(item)
```

Comment:
```suggestion
            value = [ConvertValue(item) for item in value];
```
might be clearer, but either way.

Reply:
Done